<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with anansi ‚Ä¢ anansi</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><script src="../deps/MathJax-3.2.2/tex-chtml.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Getting started with anansi">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">anansi</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.6.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/anansi.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/adjacency_matrices.html">Adjacency matrices</a></li>
    <li><a class="dropdown-item" href="../articles/differential_associations.html">Differential association testing with anansi</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/thomazbastiaanssen/anansi/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Getting started with anansi</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/thomazbastiaanssen/anansi/blob/HEAD/vignettes/anansi.Rmd" class="external-link"><code>vignettes/anansi.Rmd</code></a></small>
      <div class="d-none name"><code>anansi.Rmd</code></div>
    </div>

    
    
<style>
body {
text-align: justify}
</style>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>anansi</code> package computes and compares the association
between the features of two ‚Äôomics datasets that are known to interact
based on a database such as KEGG. Studies including both microbiome and
metabolomics data are becoming more common. Often, it would be helpful
to integrate both datasets in order to see if they corroborate each
others patterns. All vs all association is imprecise and likely to yield
spurious associations. This package takes a knowledge-based approach to
constrain association search space, only considering metabolite-function
interactions that have been recorded in a pathway database. This package
also provides a framework to assess differential association.</p>
<p>While <code>anansi</code> is geared towards metabolite-function
interactions in the context of host-microbe interactions, it is
perfectly capable of handling any other pair of datasets where some
features interact canonically. A very early version of
<code>anansi</code> was used to generate ‚ÄúExtended Data Fig. 7‚Äù in that
paper.</p>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install and load anansi</span></span>
<span><span class="co"># devtools::install_github("thomazbastiaanssen/anansi")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thomazbastiaanssen/anansi" class="external-link">anansi</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load ggplot2 and ggforce to plot results</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggforce.data-imaginist.com" class="external-link">ggforce</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Anansi supports parallelisation through the future.apply framework. </span></span>
<span><span class="co"># You can call it like this:</span></span>
<span><span class="co"># plan(multisession)</span></span>
<span></span>
<span><span class="co"># load example data + metadata from FMT Aging study</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">FMT_data</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="data-preparation">Data preparation<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h2>
<p>The main <code>anansi</code> function expects data in the
<code>anansiWeb</code> format; Basically a list with exactly three
tables: The first table, <code>tableY</code>, should be a count table of
metabolites. The second table, <code>tableX</code>, should be a count
table of functions. Both tables should have columns as features and rows
as samples.</p>
<p>The third table should be a binary adjacency matrix with the column
names of <code>tableY</code> as rows and the column names of
<code>tableX</code> as columns. Such an adjacency matrix is provided in
the <code>anansi</code> library and is referred to as a dictionary
(because you use it to look up which metabolites interact with which
functions).</p>
<p>Though this example uses metabolites and functions,
<code>anansi</code> is able to handle any type of ‚Äôomics data, as long
as there is a dictionary available. Because of this, anansi uses the
type-naive nomenclature <code>tableY</code> and <code>tableX</code>. The
Y and X refer to the position these measurements will have in the linear
modeling framework:</p>
<p><span class="math display">\[Y \sim X \times
{\text{covariates}}\]</span></p>
<div class="section level4">
<h4 id="a-note-on-functional-microbiome-data">A note on functional microbiome data<a class="anchor" aria-label="anchor" href="#a-note-on-functional-microbiome-data"></a>
</h4>
<p>Two common questions in the host-microbiome field are ‚ÄúWho‚Äôs there?‚Äù
and ‚ÄúWhat are they doing?‚Äù. Techniques like 16S sequencing and shotgun
metagenomics sequencing are most commonly used to answer the first
question. The second question can be a bit more tricky - often we‚Äôll
need functional inference software to address them. For 16S sequencing,
algorithms like PICRUSt2 and Piphillin can be used to infer function.
For shotgun metagenomics, HUMANn3 in the bioBakery suite can be
used.<br>
All of these algorithms can produce functional count data in terms of
KEGG Orthologues (KOs). These tables can be directly plugged in to
<code>anansi</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Clean and CLR-transform the KEGG orthologue table.</span></span>
<span></span>
<span><span class="co"># Only keep functions that are represented in the dictionary.</span></span>
<span><span class="va">KOs</span> <span class="op">&lt;-</span> <span class="va">FMT_KOs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">FMT_KOs</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">anansi_dic</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Cut the decimal part off.</span></span>
<span><span class="va">KOs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">floor</a></span><span class="op">(</span><span class="va">KOs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Ensure all entires are numbers.</span></span>
<span><span class="va">KOs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">KOs</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Remove all features with &lt; 10% prevalence in the dataset.</span></span>
<span><span class="va">KOs</span> <span class="op">&lt;-</span> <span class="va">KOs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">KOs</span> <span class="op">==</span> <span class="fl">0</span>, <span class="fl">1</span>, <span class="va">sum</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">KOs</span><span class="op">)</span> <span class="op">*</span> <span class="fl">0.90</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Perform a centered log-ratio transformation on the functional count table.</span></span>
<span><span class="va">KOs.exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clr_lite.html">clr_c</a></span><span class="op">(</span><span class="va">KOs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># anansi expects samples to be rows and features to be columns.</span></span>
<span><span class="va">t1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">FMT_metab</span><span class="op">)</span></span>
<span><span class="va">t2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">KOs.exp</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="weave-a-web">Weave a webüï∏Ô∏è<a class="anchor" aria-label="anchor" href="#weave-a-web"></a>
</h2>
<p>The <code><a href="../reference/web.html">weaveWeb()</a></code> function can be used to parse the tables
that we prepared above into an <code>anansiWeb</code> object. The
<code>anansiWeb</code> format is a necessary input file for the main
<code>anansi</code> workflow. It allows <code>anansi</code> to keep
track of which features from the two input data sets should should be
considered as pairs.</p>
<p>With <code>anansi</code>, we provide a pre-built map between ko, cpd
and ec features of the KEGG database, but users can provide their own
maps. <a href="https://thomazbastiaanssen.github.io/anansi/articles/adjacency_matrices.html">See
vignette on adjacency matrices.</a></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">web</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/web.html">weaveWeb</a></span><span class="op">(</span></span>
<span>  <span class="va">cpd</span> <span class="op">~</span> <span class="va">ko</span>,</span>
<span>  tableY <span class="op">=</span> <span class="va">t1</span>,</span>
<span>  tableX <span class="op">=</span> <span class="va">t2</span>,</span>
<span>  link <span class="op">=</span> <span class="fu"><a href="../reference/kegg_link.html">kegg_link</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-anansi">Run anansiüï∑Ô∏è<a class="anchor" aria-label="anchor" href="#run-anansi"></a>
</h2>
<p>The main workspider in this package is called <code>anansi</code>.
Generally, you want to give it three arguments. First, there‚Äôs
<code>web</code>, which is an <code>anansiWeb</code> object, such as the
one we generated in the above step. Second, there‚Äôs
<code>formula</code>, which should be a formula. For instance, to assess
differential associations between treatments, we use the formula
<code>~Legend</code>, provided we have a column with that name in our
<code>metadata</code> object, the Third argument.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anansi_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/anansi.html">anansi</a></span><span class="op">(</span></span>
<span>  web <span class="op">=</span> <span class="va">web</span>,</span>
<span>  formula <span class="op">=</span> <span class="op">~</span><span class="va">Legend</span>,</span>
<span>  metadata <span class="op">=</span> <span class="va">FMT_metadata</span>,</span>
<span>  adjust.method <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Fitting least-squares for following model:</span></span>
<span><span class="co">## ~ x + Legend + x:Legend</span></span></code></pre>
<pre><code><span><span class="co">## Running correlations for the following groups:</span></span>
<span><span class="co">##  Aged yFMT, Aged oFMT, Young yFMT</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="spin-to-a-table">Spin to a tableüìù<a class="anchor" aria-label="anchor" href="#spin-to-a-table"></a>
</h2>
<p><code>anansi</code> returns a wide format table as an output. For
general reporting, we recommend sticking to the table format as it‚Äôs the
most legible.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use tidyr to wrangle the correlation r-values to a single column</span></span>
<span><span class="va">anansiLong</span> <span class="op">&lt;-</span> <span class="va">anansi_out</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"All"</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">"FMT"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate_wider_delim.html" class="external-link">separate_wider_delim</a></span><span class="op">(</span><span class="va">name</span>, delim <span class="op">=</span> <span class="st">"_"</span>, names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cor_group"</span>, <span class="st">"param"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">param</span>, values_from <span class="op">=</span> <span class="va">value</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Now it's ready to be plugged into ggplot2, though let's clean up a bit more.</span></span>
<span><span class="co"># Only consider interactions where the entire model fits well enough.</span></span>
<span><span class="va">anansiLong</span> <span class="op">&lt;-</span> <span class="va">anansiLong</span><span class="op">[</span><span class="va">anansiLong</span><span class="op">$</span><span class="va">full_q.values</span> <span class="op">&lt;</span> <span class="fl">0.2</span>, <span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="plot-the-results">Plot the results<a class="anchor" aria-label="anchor" href="#plot-the-results"></a>
</h2>
<p>The long format can be helpful to plug the data into
<code>ggplot2</code>. Here, we recreate part of the results from the FMT
Aging study.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">anansiLong</span><span class="op">)</span> <span class="op">+</span></span>
<span></span>
<span>  <span class="co"># Define aesthetics</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="va">r.values</span>, y <span class="op">=</span> <span class="va">feature_X</span>,</span>
<span>    fill <span class="op">=</span> <span class="va">cor_group</span>, alpha <span class="op">=</span> <span class="va">disjointed_Legend_p.values</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span></span>
<span>  <span class="co"># Make a vertical dashed red line at x = 0</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span></span>
<span>  <span class="co"># Points show  raw correlation coefficients</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">21</span>, size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span></span>
<span>  <span class="co"># facet per compound</span></span>
<span>  <span class="fu">ggforce</span><span class="fu">::</span><span class="fu"><a href="https://ggforce.data-imaginist.com/reference/facet_row.html" class="external-link">facet_col</a></span><span class="op">(</span><span class="op">~</span><span class="va">feature_Y</span>, space <span class="op">=</span> <span class="st">"free"</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span></span>
<span>  <span class="co"># fix the scales, labels, theme and other layout</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_y_discrete</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_alpha_manual</a></span><span class="op">(</span></span>
<span>    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRUE"</span> <span class="op">=</span> <span class="fl">1</span>, <span class="st">"FALSE"</span> <span class="op">=</span> <span class="fl">1</span> <span class="op">/</span> <span class="fl">3</span><span class="op">)</span>,</span>
<span>    <span class="st">"Disjointed association\np &lt; 0.05"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span></span>
<span>    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>      <span class="st">"Young yFMT"</span> <span class="op">=</span> <span class="st">"#2166ac"</span>,</span>
<span>      <span class="st">"Aged oFMT"</span> <span class="op">=</span> <span class="st">"#b2182b"</span>,</span>
<span>      <span class="st">"Aged yFMT"</span> <span class="op">=</span> <span class="st">"#ef8a62"</span>,</span>
<span>      <span class="st">"All"</span> <span class="op">=</span> <span class="st">"gray"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Young yFMT"</span>, <span class="st">"Aged oFMT"</span>, <span class="st">"Aged yFMT"</span>, <span class="st">"All"</span><span class="op">)</span>, <span class="st">"Treatment"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Pearson's \u03c1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="anansi_files/figure-html/plot_FMT-1.png" alt="Differential association plot, with facets separating metabolites 
and metabolites on the y-axis. x-axis depicts pearson's correlation 
coefficient." width="960"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: TUXEDO OS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.12.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Amsterdam</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] tidyr_1.3.1      ggforce_0.4.2    ggplot2_3.5.1    anansi_0.6.0    </span></span>
<span><span class="co">## [5] BiocStyle_2.32.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] SummarizedExperiment_1.34.0 gtable_0.3.6               </span></span>
<span><span class="co">##  [3] xfun_0.50                   bslib_0.9.0                </span></span>
<span><span class="co">##  [5] htmlwidgets_1.6.4           Biobase_2.66.0             </span></span>
<span><span class="co">##  [7] lattice_0.22-6              vctrs_0.6.5                </span></span>
<span><span class="co">##  [9] tools_4.4.2                 generics_0.1.3             </span></span>
<span><span class="co">## [11] stats4_4.4.2                parallel_4.4.2             </span></span>
<span><span class="co">## [13] tibble_3.2.1                pkgconfig_2.0.3            </span></span>
<span><span class="co">## [15] Matrix_1.7-2                desc_1.4.3                 </span></span>
<span><span class="co">## [17] S4Vectors_0.42.1            lifecycle_1.0.4            </span></span>
<span><span class="co">## [19] GenomeInfoDbData_1.2.12     stringr_1.5.1              </span></span>
<span><span class="co">## [21] farver_2.1.2                compiler_4.4.2             </span></span>
<span><span class="co">## [23] textshaping_1.0.0           munsell_0.5.1              </span></span>
<span><span class="co">## [25] codetools_0.2-20            GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">## [27] htmltools_0.5.8.1           sass_0.4.9                 </span></span>
<span><span class="co">## [29] yaml_2.3.10                 pkgdown_2.1.1              </span></span>
<span><span class="co">## [31] pillar_1.10.1               crayon_1.5.3               </span></span>
<span><span class="co">## [33] jquerylib_0.1.4             MASS_7.3-64                </span></span>
<span><span class="co">## [35] DelayedArray_0.30.1         cachem_1.1.0               </span></span>
<span><span class="co">## [37] abind_1.4-8                 parallelly_1.42.0          </span></span>
<span><span class="co">## [39] tidyselect_1.2.1            digest_0.6.37              </span></span>
<span><span class="co">## [41] stringi_1.8.4               future_1.34.0              </span></span>
<span><span class="co">## [43] purrr_1.0.4                 dplyr_1.1.4                </span></span>
<span><span class="co">## [45] bookdown_0.42               listenv_0.9.1              </span></span>
<span><span class="co">## [47] labeling_0.4.3              polyclip_1.10-7            </span></span>
<span><span class="co">## [49] fastmap_1.2.0               grid_4.4.2                 </span></span>
<span><span class="co">## [51] colorspace_2.1-1            cli_3.6.4                  </span></span>
<span><span class="co">## [53] SparseArray_1.4.8           magrittr_2.0.3             </span></span>
<span><span class="co">## [55] S4Arrays_1.4.1              future.apply_1.11.3        </span></span>
<span><span class="co">## [57] withr_3.0.2                 scales_1.3.0               </span></span>
<span><span class="co">## [59] UCSC.utils_1.0.0            rmarkdown_2.29             </span></span>
<span><span class="co">## [61] XVector_0.44.0              httr_1.4.7                 </span></span>
<span><span class="co">## [63] matrixStats_1.5.0           globals_0.16.3             </span></span>
<span><span class="co">## [65] igraph_2.1.4                ragg_1.3.3                 </span></span>
<span><span class="co">## [67] evaluate_1.0.3              knitr_1.49                 </span></span>
<span><span class="co">## [69] GenomicRanges_1.56.2        IRanges_2.38.1             </span></span>
<span><span class="co">## [71] MultiAssayExperiment_1.33.1 rlang_1.1.5                </span></span>
<span><span class="co">## [73] Rcpp_1.0.14                 glue_1.8.0                 </span></span>
<span><span class="co">## [75] tweenr_2.0.3                BiocManager_1.30.25        </span></span>
<span><span class="co">## [77] BiocGenerics_0.50.0         rstudioapi_0.17.1          </span></span>
<span><span class="co">## [79] jsonlite_1.9.0              R6_2.6.1                   </span></span>
<span><span class="co">## [81] MatrixGenerics_1.16.0       systemfonts_1.2.1          </span></span>
<span><span class="co">## [83] fs_1.6.5                    zlibbioc_1.50.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Thomaz Bastiaanssen, Thomas Quinn, Giulio Benedetti, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
